Receipt/Sources/ListReceiptPresenter.swift

```diff
@@ -50,22 +52,98 @@ final class ListReceiptPresenter {
+                                            receiptSourceType: .user)
+                   self.segmentedControlItems.append(segmentedControlItem)
+                   self.selectedSegmentControlItem = segmentedControlItem
+            }
+            self.prepaidCardRepository
+                    .listPrepaidCards(queryParam: self.setUpPrepaidCardQueryParam()) { [weak self] (result) in
+                guard let strongSelf = self, let view = strongSelf.view else {
+                    return
+                }
+                switch result {
+                case .success(let pageList):
+                    if let prepaidCards = pageList?.data {
+                        prepaidCards.filter { $0.primaryCardToken == nil }.forEach {
+                            strongSelf.segmentedControlItems.append(
+                                SegmentedControlItem(
+                                    token: $0.token ?? "",
+                                    segmentedControlHeader: $0.formattedCardBrandCardNumber ?? "",
+                                    receiptSourceType: .prepaidCard))
+                        }
+                        prepaidCards.filter { $0.primaryCardToken != nil }.forEach {
+                            strongSelf.segmentedControlItems.append(
+                                SegmentedControlItem(
+                                    token: $0.token ?? "",
+                                    segmentedControlHeader: $0.formattedCardBrandCardNumber ?? "",
+                                    receiptSourceType: .prepaidCard))
+                        }
+                    }
+                    if strongSelf.selectedSegmentControlItem == nil {
+                        strongSelf.selectedSegmentControlItem = strongSelf.segmentedControlItems[0]
+                    }
+
+                case .failure(let error):
+                    view.showError(error, pageName: strongSelf.pageName, pageGroup: strongSelf.pageGroup) {
+                        strongSelf.listAllAvailableReceipts()
+                    }
+                    return
+                }
+                    }
+            self.loadReceiptsForSelectedToken()
+        }
+    }
+
+    func loadReceiptsForSelectedToken() {
+        sectionData.removeAll()
+        if let selectedItem = selectedSegmentControlItem {
+            if selectedItem.receiptSourceType == .user {
+                offset = 0
+                listUserReceipts()
+            } else {
+                listPrepaidCardReceipts(selectedItem.token)
+            }
+        }
     }
 
     func listReceipts() {
-        if let prepaidCardToken = prepaidCardToken {
-            listPrepaidCardReceipts(prepaidCardToken)
+        if showAllAvailableSources {
+            listAllAvailableReceipts()
         } else {
-            listUserReceipts()
+            if let prepaidCardToken = prepaidCardToken {
+                listPrepaidCardReceipts(prepaidCardToken)
```

Comment:
no need of this method anymore, you can reuse loadReceiptsForSelectedToken

Reply:
Do you mean get rid of `listPrepaidCardReceipts`? Not sure I follow